File: Basics_with_JavaScript/1314 Nested Loops/14 Nested loops Excercise/02. Equal Sums Even Odd Position.js
function vladi(array) {
  let startNumber = Number(array[0]);
  let endNumber = Number(array[1]);
  let oddSum = 0;
  let evenSum = 0;
  let currentNum = 0;
  let currentNumString = "";
  let currentDigit = 0;
  let magicLine = "";

  for (let i = startNumber; i <= endNumber; i++) {
    currentNum = i;
    currentNumString = "" + currentNum;

    for (let j = 0; j < currentNumString.length; j++) {
      let currentDigitString = currentNumString.charAt(j);
      currentDigit = Number(currentDigitString);

      if (j % 2 === 0) {
        oddSum += currentDigit;
      } else {
        evenSum += currentDigit;
      }
    }

    if (oddSum == evenSum) {
      magicLine += currentNum + " ";
    }

    oddSum = 0;
    evenSum = 0;
  }
  console.log(magicLine);
}


vladi(["100000",
"100050"])
;
vladi(["123456",
"124000"])
;
vladi(["299900",
"300000"])
;
vladi(["100115",
"100120"])
;

File: Basics_with_Python/PythonBasicCourse/6_Nested_Loops/14_Nested_Loops___Exercise/02._Equal_Sums_Even_Odd_Position.py
start_number = int(input())
end_number = int(input())

for number in range(start_number, end_number + 1):
    str_number = str(number)
    odd_sum = 0
    even_sum = 0
    for ind, num in enumerate(str_number):
        if ind % 2 == 0:
            odd_sum += int(num)
        else:
            even_sum += int(num)
    if odd_sum == even_sum:
        print(number, end=" ")
    else:
        continue
